### ISSUE TITLE
Remapper is not working with PS4 Remote Play (Version 2.5.0.9220)

### ISSUE THREAD

**Issue by MiWo85** [komefai/PS4Macro#8]
Dear developer, 

I've downloaded the latest 0.5.1 BETA Version of PS4Macro because I wanted to try the keyboard mappings within PS4 Remote Play (Version 2.5.0.9220).

I opened PS4 Remote Play first and started the PS4Macro afterwards. I opened the Remapper Tool as you described it in your wiki and focused the PS4 Remote Play Windows afterwards again an started the session to my PS4 by clicking on "Start". I further tried to open the PS4Macro programm after the connection to my Playstation 4 was established, but same issue. Just the common keys are working (ENTER for X, ESX for Return, Left/Right Arrow Key). 

Another step was to open PS4Macro with following configuration inside the settings.xml:

_<?xml version="1.0" encoding="utf-8"?>
<Settings>
 <AutoInject>true</AutoInject>
  <EmulateController>true</EmulateController>
  <ShowConsole>true</ShowConsole>
</Settings>_

which gives me an "OnInjectionSuccess 7384". I guess this should be the process ID of the PS4 Remote Play Instance, but I'm still not able to use the extended key mapping (like R1 with E or R2 with O).

At the moment I have no PS4 Controller connected to my USB because as far as I understood this is not neccesary when using the "EmulateConntroller" Option?

Do I have any understanding trouble or is there any bug?

If you need further information, don't hesitate to ask me again.

Thank you very much in advance,
Michael

P.S.: I tried to use the Mapping Keys within the FIFA 18 Main Menu.

**Comment by komefai** [comment]
Hi! Thanks for the details. You're right about not needing a PS4 Controller for `EmulateController` option.

Please close the programs and try this instead:

1. Run PS4 Remote Play without PS4 controller
2. Run PS4 Macro
3. Wait for this screen

![ps4_emulate](https://user-images.githubusercontent.com/30201667/34679571-f66d4744-f4c8-11e7-82b3-a7035f5aa3b3.png)

4. Press start
5. Go to Tools->Remapper to start the remapper

Let me know if it fixes the problem or not.

**Comment by MiWo85** [comment]
Good Morning komefai,

I exactly did the the steps you mentioned yesterday and again right now. (with emulatecontroller = true option). OnInjectionSuccess is printed again. 

But I'm still not able to use any mapped key (just ESCAPE, ENTER and Arrow left, right, up, down).

I tried it inside FIFA 18 and the Playstation Main Menu. 

My operating system is Windows 8 64bit inside a VMware Player, but I think this should be no problem. 
I'm using Ps4 Remote Play over WAN at the moment and not on a computer in my home network. Could this cause the problems?

Thank you very much for your fast response.
Best Regards

**Comment by komefai** [comment]
Using it in a VM and over WAN shouldn't be a problem. However, you mentioned that: 

> But I'm still not able to use any mapped key (just ESCAPE, ENTER and Arrow left, right, up, down).

This means that PS4 Macro are not intercepting the keys at all and letting you use the actual keys. So make sure that you're running PS4Macro.exe as Administrator and make sure that the Remapper window (Tools->Remapper) is open at all times when you're using it for remapping the keys.

**Comment by MiWo85** [comment]
I gave it another try today with the version 0.5.1.

1) I opened PS4 Remote Play without Controller (as local Administrator)
2) The following Screen appears:
![ps4macro1](https://user-images.githubusercontent.com/33148913/35227316-15f3ba10-ff8e-11e7-9705-a5c5e2c56332.png)
3) I open PS4Macro.exe with local Administrator Privileges
![ps4macro2](https://user-images.githubusercontent.com/33148913/35228131-3f4d338a-ff90-11e7-9220-1700b7a5ba61.png)
4) InjectionSucced Message Appears

5) Configuration of Settings.xml:
_<?xml version="1.0" encoding="utf-8"?>
<Settings>
  <AutoInject>true</AutoInject>
  <EmulateController>true</EmulateController>
  <ShowConsole>true</ShowConsole>
</Settings>_

6) Establishing the Connection to my PS4 and opened Tools -> Remapper **afterwards**. Both Applications are open. I try to use the A S D W keys on my keyboard for left, down, right, up but nothing happens. 

![ps4macro3](https://user-images.githubusercontent.com/33148913/35228744-c07cc03c-ff91-11e7-840f-ffd3ee331197.png)

So any idea / advice what is my problem or fault?

**Comment by leguims** [comment]
There is a new version 0.5.2 . Try with this one.

**Comment by MiWo85** [comment]
It's the same problem with 0.5.2.

**Comment by komefai** [comment]
Hi, Your latest RemotePlay screenshots does not detect the emulated controller yet. You have to wait for this screen with this exact message otherwise the emulation will not work: “Press the OPTIONS button on the controller to start.”

![img](https://user-images.githubusercontent.com/30201667/34679571-f66d4744-f4c8-11e7-82b3-a7035f5aa3b3.png)

**Comment by MiWo85** [comment]
So it seems that the PS4 Remote Play is not detecting the emulated controller?
Just to be sure, I have no physical controller connected yet and the emulated controller is coming within the PS4Macro by the <EmulateController> tag within the settings.xml?

**Comment by komefai** [comment]
Yes you’re correct, all DualShock4 controllers should be unplugged from your PC when you’re using EmulateContoller.

**Comment by MiWo85** [comment]
Okay, I've never connected a DS4 Controller to my System yet. So I don't know why the emulation is not working at all =). Any Suggestions?

**Comment by MiWo85** [comment]
Short Update:

Yesterday I connected my PS4 Controller to my laptop and started the PS4 Remote Play. I got this window / message:

![grafik](https://user-images.githubusercontent.com/33148913/35377042-fda5ead8-01ad-11e8-8200-77488ba686f6.png)

I further started the PS4Macro and opened the Remapper. I was able to navigate with the ASDW Buttons to the first time.

Afterwards I closed all programms and unplugged the physical controller and did another test with the emulatedController Tag within my settings.xml.

The emulated Controller **was not** recognized by the PS4 Remote Play and thus the Remapper Keys were not working. So from my point of view it seems that there might be a problem / bug with the emulatedController option as I tested this feature on two different machines (one physical and one VM) and on both it is not working.

**Comment by komefai** [comment]
Thanks for the detailed report. I'm going to reopen this issue and see what I can do with it.

**Comment by Y-T-G** [comment]
AutoInject: false fixed the same issue for me.

**Comment by MiWo85** [comment]
Can confirm partially.

Tried the following constellation:

<?xml version="1.0" encoding="utf-8"?>
<Settings>
  <AutoInject>false</AutoInject>
  <BypassInjection>true</BypassInjection>
  <EmulateController>true</EmulateController>
  <ShowConsole>true</ShowConsole>
</Settings>

EmulatedController was not recognized.

I had to false every setting inside the settings.xml except the EmulatedController Tag. 
Afterwards the emulatedController was detected in PS4 Remote Play. Opened Remapper before I connected to my Playstation, but the remapped keys still not working =)

**Comment by komefai** [comment]
@MiWo85 I can confirm that opening Remapper before pressing "Start" and then pressing Z on your keyboard to start **will cause it to NOT work 100% of the time.** This is probably because PS4 Macro detects and cache the start window (for performance reasons) thinking that it was the game window.

What you should do is wait for the screen to make sure that the controller is emulated and then **CLICK** the "Start" button. Wait for the game before you open the Remapper window.

**Comment by MiWo85** [comment]
@komefai 
I tried ~ 20 times it in various constellations since yesterday. I've downloaded the latest version with Remapper Mouse Support.

Yesterday it worked one time for me this way:
1) Opened PS4 Remote
2) Opened PS4Macro
3) Opened Remapper
4) Clicked "Mouse Support"
5) Clicked on PS4 Remote Window an started Session with Z-Button.

Afterwards I was able to navigate ingame with the customized keys and it worked very good. But all the other tries did not work. 

Just tried your recommendation to START first (after emulated Controller was detected) and waited to open the Remapper Tool after the Playstation Screen was shown. But the injection of the remapper keys did not work. 

The one succesfull scenario yesterday showed that it might work randomly, but not reliable. Is there anything I can do for you to help you finding the reason for this?

**Comment by komefai** [comment]
I just saw your settings from your previous post:

```
<?xml version="1.0" encoding="utf-8"?>
<Settings>
  <AutoInject>false</AutoInject>
  <BypassInjection>true</BypassInjection>
  <EmulateController>true</EmulateController>
  <ShowConsole>true</ShowConsole>
</Settings>
```

..and setting BypassInjection to true is one of the reasons why it doesn't work. Please try everything again.

**Comment by MiWo85** [comment]
Please read my post again two days ago:

> I had to false every setting inside the settings.xml except the EmulatedController Tag.
> Afterwards the emulatedController was detected in PS4 Remote Play. Opened Remapper before I connected to my Playstation, but the remapped keys still not working.

So everything is already on false except the EmulatedController Tag. The Controller is also detected everytime when I open PS4 Remote Play and PS4 Macro.

**Comment by ninjatjj** [comment]
I don't know if you want "me too" comments but I can also say that when I have everything on false except "EmulateController" it does not work for me. Occasionally I think something is slightly different as remote play does not accept arrow keys but never got "Z" or "WASD" to work.

Timing issue?

**Comment by komefai** [comment]
@MiWo85 
Whoops sorry, my bad.

@ninjatjj 
Thank you for the report!

One easy way to make sure that the emulated controller is actually working is to try the macro recording. You should see the red numbers increasing every frame on the bottom right corner if it's working properly. I suggest anyone with this problem to try this instead of the Remapper because this method has a better visual feedback, which is easier for us to debug.

![Macro Recording Test](https://user-images.githubusercontent.com/30201667/35946283-c17840f4-0c95-11e8-99bd-6398ba212b6b.png)

**Comment by Y-T-G** [comment]
Well, this works for me most of time (other times, the mouse doesn't work but keyboard works always):
1. Start Remote Play App
2. Start PS4Macro as Administrator.(@komefai  I have observed that if you don't start it as administrator the keys become  permanently bugged even after closing the remote play window and remapper)
3. Click Start in Remote Play App after you see controller detected screen.
4. After the Remote Play App starts displaying your PS4's screen, start remapper and click on the Remote Play window.

Here's my settings.xml for the sake of clarity:
`<?xml version="1.0" encoding="utf-8"?>
<Settings>
  <AutoInject>false</AutoInject>
  <BypassInjection>false</BypassInjection>
  <EmulateController>true</EmulateController>
  <ShowConsole>false</ShowConsole>
</Settings>
`

**Comment by ninjatjj** [comment]
Unfortunately this isn't working for me following @Y-T-G instructions. TBH though I can't see Enable Mouse - I am using 0.5.2 is there a better release to try?

**Comment by Y-T-G** [comment]
@ninjatjj Yes, there are debug builds here: https://github.com/komefai/PS4Macro/issues/13 .  Although latest version has a bug. Use previous version. Use the slightly updated instruction above again.

**Comment by MiWo85** [comment]
@ninjatjj

For me neither. I tried YTGs instructions several times. I did try any other possible and mentioned way with different PS4Macro versions but out of 50 tries just one time it worked.

**Comment by Y-T-G** [comment]
@MiWo85 Did you try again with the updated instructions?

**Comment by ninjatjj** [comment]
This has confused me for a long time but when it says "some machines don't support autoinject" I really don't understand what it means? I am not really using any peripheral here so why would my machine need to autoinject something - this is windows 10. Perhaps it relates to the macro aspect rather than the keyboard input?

Also, do you have to have plugged in a controller at least once?

Is it easy to set up in the IDE? If so I could try to debug here. Where would I see "Z" getting mapped to a button on the controller?

Does it matter if you are using a built in laptop keyboard rather than a USB keyboard?

**Comment by komefai** [comment]
@ninjatjj AutoInject is a feature that allows PS4 Macro to intercept PS4 controls multiple times without having to start a clean instance of Remote Play each time you open PS4 Macro. From reports in the past, some PCs just don't support this feature and throws and error for an unknown reason. So it is the best to set it to false if you're unsure that your PC is supported or not (until the issue is solved).

> Also, do you have to have plugged in a controller at least once?

I haven't tested this theory myself but it is possible.

> Is it easy to set up in the IDE? If so I could try to debug here.

There are many online resources on setting up Visual Studio so I think it's not too difficult.

> Where would I see "Z" getting mapped to a button on the controller?

You can see it in the `Mappings` list in the Remapper window. Z is mapped to the `OPTIONS` button by default.

> Does it matter if you are using a built in laptop keyboard rather than a USB keyboard?

No, it does not matter.

**Comment by MiWo85** [comment]
@Y-T-G 
Yes I also did the updated insutructions in three different tries. 

The "Z" Button and Mouse Input is working for me until the Connection to my PS4Remote is established. But afterwards it does not work anymore.

**Comment by Y-T-G** [comment]
@MiWo85 Can you try Registering your PS4 again without signing in with your PSN ID in the Remote Play application?

**Comment by komefai** [comment]
I'm working on a status checking tool that will hopefully help us pinpoint the cause of this problem. I'll compile and post the debug build here soon.

**Comment by JXNwarrior** [comment]
I am new to programming and coding but I am having trouble finding the file for "settings.xml" forgive me if it seems obvious

**Comment by komefai** [comment]
@JXNwarrior You can create one yourself using any text editor like notepad.

https://github.com/komefai/PS4Macro#settings

**Comment by komefai** [comment]
@MiWo85 @ninjatjj 

### Update

My most recent commit added "Status Checker" window which you can find in `Help->Status Checker`. This window should contain enough information to pinpoint this issue (or any other issues in the future).

![ps4macro-statuschecker](https://user-images.githubusercontent.com/30201667/36349639-9fc7a33e-14bd-11e8-898f-395e02638e2d.png)

### Solution to this issue?

While I was developing this feature I also found out that C# `Process` objects are using a snapshot of the process. So a [refresh](https://msdn.microsoft.com/en-us/library/system.diagnostics.process.refresh(v=vs.110).aspx) is required when Remote Play changes from the startup window into the game streaming window.

This means that this latest build should hopefully fix this issue, whether you are starting Remapper and pressing Z to start, or start by clicking on the Start button on Remote Play. **It shouldn't matter as long as you close and reopen Remapper once AFTER the game starts streaming to perform a [refresh](https://msdn.microsoft.com/en-us/library/system.diagnostics.process.refresh(v=vs.110).aspx).**

---

## DOWNLOAD

[PS4Macro_c43dbd4.zip](https://github.com/komefai/PS4Macro/files/1734519/PS4Macro_c43db4.zip) - [commit [c43dbd4](https://github.com/komefai/PS4Macro/commit/c43dbd4719ee5e31ca54989f82d19f0e1b8389d5)]

**Comment by JXNwarrior** [comment]
I am getting an error saying that the interceptor is not working. I have also created a settings.xml with all false commands except for emulate controller.

**Comment by Y-T-G** [comment]
@JXNwarrior Did you start the Remote Play application before running PS4Macro?

**Comment by JXNwarrior** [comment]
yes. I have started PS4Macro after the inital "click to start" screen and I have tried waiting until the PS4 connected to start it. neither worked

**Comment by Y-T-G** [comment]
Can you enable ShowConsole in settings.xml and tell what do you see in the console after starting PS4Macro?

**Comment by JXNwarrior** [comment]
it says "OnInjectionSuccess 5396"

**Comment by Y-T-G** [comment]
Upload a screenshot of your screen while keeping the remote play tab focused(keeping the status checker window beside the remote play window).

**Comment by JXNwarrior** [comment]
![scrn](https://user-images.githubusercontent.com/36576573/36355770-ebafe4ba-14ad-11e8-84f3-27e2c3926f72.jpg)

**Comment by Y-T-G** [comment]
You did not properly setup settings.xml, I guess. For your ease, download the attached one and replace yours with this one.

**Comment by JXNwarrior** [comment]
same thing happened again. is there something else that I should be doing with settings.xml other than having it in the same folder? 
![scrn2](https://user-images.githubusercontent.com/36576573/36355937-90c9605a-14b0-11e8-9829-9123924b35c4.jpg)

**Comment by Y-T-G** [comment]
If you use an antivirus, try adding the folder to exclusions.

**Comment by Y-T-G** [comment]
No, there is nothing else to do with settings.xml. Komefai would know what is wrong better than me.

**Comment by ninjatjj** [comment]
I have the same diagnostic from Help->Status Checker - it shows the interceptor is not working

**Comment by komefai** [comment]
@JXNwarrior @ninjatjj Can you try running Status Checker on the Remote Play startup screen without starting the game stream and upload a screenshot. Does anything happen when you press 'Z' on Remote Play while Remapper is open?

**Comment by JXNwarrior** [comment]
![capture](https://user-images.githubusercontent.com/36576573/36826088-b2c324a2-1cd0-11e8-8a3f-578c144c9808.JPG)
 
Here's the screenshot, and no nothing happens when I open the remapper and press 'Z' or any other button.

**Comment by ninjatjj** [comment]
Same here :'(

How are you detecting "Not Working" for the "Interceptor" - can you provide a link to the source code where that check is done that please?

**Comment by Y-T-G** [comment]
@ninjatjj You might have missed missed komefai's updated build. Here's the link to the download https://github.com/komefai/PS4Macro/files/1734519/PS4Macro_c43db4.zip

**Comment by Y-T-G** [comment]
@ninjatjj https://github.com/komefai/PS4Macro/blob/c43dbd4719ee5e31ca54989f82d19f0e1b8389d5/PS4Macro/Classes/StatusChecker.cs

**Comment by ninjatjj** [comment]
That is the version I am using. I should have been clearer, I can see the same problem "Not Working" for the "Interceptor". What I was wondering is where in the code the check is that @komefai does to determine if it is working or not so try to work out what might be going wrong.

**Comment by ninjatjj** [comment]
Yeah, I see that now. https://github.com/komefai/PS4Macro/blob/c43dbd4719ee5e31ca54989f82d19f0e1b8389d5/PS4Macro/Classes/StatusChecker.cs#L169

I can't really see which of these is actually calling OnReceiveData https://github.com/komefai/PS4Macro/search?utf8=%E2%9C%93&q=OnReceiveData&type=

**Comment by ninjatjj** [comment]
oh, it is probably MainForm.cs, something to do with the Interceptor

**Comment by Y-T-G** [comment]
@ninjatjj Going through the code, I think it shows 'Working' state, if it finds out that the Remote Play application has detected a Dualshock 4 (Emulated in this case). Is that right @komefai ?

**Comment by Y-T-G** [comment]
@ninjatjj Going through the code, I think Interceptor shows 'Working' state, if it finds that Remote Play application has detected a DualShock 4. Is that right @komefai ?

**Comment by komefai** [comment]
@Y-T-G That is correct. Technically speaking, the status changes to "Working" when the hooked method `OnReceiveData` gets called by Remote Play. Same goes to the keyboard and mouse status.

**Comment by Y-T-G** [comment]
Does [this solution](https://github.com/komefai/PS4Macro/issues/17#issuecomment-373500212) work for anyone else ?

**Comment by mayconramos21** [comment]
Do you have any version for macbook @komefai ?

**Comment by tsikerdekis** [comment]
I am having the same issue. Has this been resolved?

**Comment by mayconramos21** [comment]
@tsikerdekis I'm using a program called parallels, which will simulate win8.

**Comment by hamslicejumbomice** [comment]
@komefai How am I suppose to record a macro using my keyboard if i have to have the remapper open?The entire reason I wanted to use my keyboard was to record a macro to use with my playstation. I would use the controller to record the macro but it would be harder. Is there a way to record the macro with the remapping open? 

It seems that I can import a macro while the remapper is open but I can't do that until I record my macro first! I would program one but I am not sure how to do that! Also what is this settings.xml file everyone is talking about? I looked everywhere for it and couldn't find it but thats okay because I got PS4Macro to work just fine without changing any configurations

**Comment by tsikerdekis** [comment]
I have tried different configurations but to no avail. I have win10 and I have been using PS4Macro 0.5.2, 0.5.1, c43db4, 658584b.

Settings are false, false, true, false. I use the settings.xml posted by @Y-T-G .

I noticed that no emulated controller appears on the device drivers when running ps4macro but if I plug in DS4 then I have a generic game controller driver appears.

I also noticed that if I run ps4macro close it and then attempt to run another version of ps4macro it says that the remote play has been already injected by another ps4macro executable.

Interceptor is not working and I never get the ps4 remote controller screen saying that the controller is connected (and allowing to click options).

**Comment by KnittingQueen123** [comment]
Id love some help on my end please. Ive followed this thread and many more. Heres my details so far
![image](https://user-images.githubusercontent.com/40581786/41878109-9e5681f8-7888-11e8-9c3a-9d60c6c68ca9.png)

**Comment by grill2010** [comment]
I don't know if this issue is related to the same bug/ limitation I have encountered in my app. I'm the author of [PSJoy ](http://grill2010.github.io/psJoy.html) and I got already in contact with @komefai long time ago, I was asking him about intercepting the Remote Play screen.
Recently some customers of my app wrote me that the emulated Dualshock gamepad is not working and that it does not process inputs. It took me a while to figure out what is going wrong but the reason is simple. All users which suffered from this strange issue didn't have an HID device connected to their PC as they did use an ordinary laptop. The PS4Macro also hooks into the **CreateFileW** and is returning a dummy handle whenever an HID device handle is requested. However, as long as no HID device is connected to the PC this method never gets called, at least not with an HID filename.
So I thought how can I make the PS4 Remote Play program think an HID device is connected so that the **CreateFileW** gets called with an HID parameter. The solution is to also hook into the _setupapi.dll_ methods like [**SetupDiEnumDeviceInterfaces**](https://docs.microsoft.com/de-de/windows/desktop/api/setupapi/nf-setupapi-setupdienumdeviceinterfaces) and [**SetupDiGetDeviceInterfaceDetailW**](https://docs.microsoft.com/de-de/windows/desktop/api/setupapi/nf-setupapi-setupdigetdeviceinterfacedetailw). I got it to work a few hours ago but I still need to test it but so far it looks good.
I believe that some of these issues in this thread are also related to that problem. @komefai if you are interested in to solve that problem just let me know if you want I can also make a pull request but I don't know when I will have time.

**Comment by eglyants** [comment]
@grill2010 @komefai grill2010, were you able to find a solution to this?  I'm trying to run the setup on AWS EC2 and believe HID is restricting remapping.

**Comment by grill2010** [comment]
Edit: seems that eglyants deleted his/ her questions so I guess it was just a misunderstanding or his/ her questions was related to another issue.

**Comment by eglyants** [comment]
@grill2010 Thank you for your reply. I'm sure I have the same problem because I'm attempting to run it on aws ec2.  Is there a dummy HID driver you could suggest?

**Comment by grill2010** [comment]
Not really, you can try vJoy which is a virtual HID gamepad driver but I guess just installing a virtual driver to make it work is a little bit over over the top. Could you not just connect a mouse or a keyboard to your PC/ Laptop?

**Comment by jbsoccerbrit** [comment]
> Id love some help on my end please. Ive followed this thread and many more. Heres my details so far
> ![image](https://user-images.githubusercontent.com/40581786/41878109-9e5681f8-7888-11e8-9c3a-9d60c6c68ca9.png)

UAC was a killer to me, once I disabled it, it helped.
Also make sure the main .exe isn't blocked by right clicking on it and going to properties.
I flagged it also to run as administrator and download the provided settings.xml

This got me past the issues you are having.

**Comment by abstractoakbow** [comment]
![image](https://user-images.githubusercontent.com/46252000/50543185-24af9f00-0b96-11e9-8fde-ee544965c107.png)
Can confirm that @grill2010 indeed solved my problem. Couldn't figure out why only interceptor didn't work. Plugged a mouse into my laptop, and presto, interceptor works. That being said, I can't get it to work with the official 0.52 build, but that doesn't particularly bother me. And finally, I copy and pasted my "bindings" from 0.52 to cf7ce2c. Every time I open the cf7ce2c, I get this error, but I don't see any problems. I can attempt to reproduce my findings, but without importing my previous 0.52 bindings if anyone's curious.
TLDR: if interceptor isn't working on laptop, try plugging a usb device in until this problem is fixed. (along with using cf7ce2c version, and using the settings provided)

**Comment by defjamk12** [comment]
I was having the same problem with interceptor not working (on my laptop) but I found a workaround which worked for me and may help others.

Issue Reproduced:
![image](https://user-images.githubusercontent.com/46227435/50544531-68ba9800-0bc6-11e9-90b3-a8969512769e.png)

Workaround Steps:
1) Close PS4Macro and PS4 Remote Play if they're running
2) Turn Bluetooth 'Off'
![image](https://user-images.githubusercontent.com/46227435/50544603-863d3100-0bc9-11e9-8d78-de8296b5a41c.png)

3) Run PS4 Remote Play
4) Run PS4Macro (I use cf7ce2c for this example)
5) Turn Bluetooth 'On'
![image](https://user-images.githubusercontent.com/46227435/50544623-3a3ebc00-0bca-11e9-9207-a146ec972a4a.png)

6) Interceptor now shows as 'Working' and PS4 Remote Play now shows “Press the OPTIONS button on the controller to start.”
![image](https://user-images.githubusercontent.com/46227435/50544629-a7525180-0bca-11e9-841e-6294f47f677f.png)

7) Click 'Start'
8) Go to PS4Macro and go to 'Tools' -> 'Remapper'
9) Go to PS4 Remote Play and now the keyboard mappings should work for games

This is my settings.xml

`<?xml version="1.0" encoding="utf-8"?>
<Settings>
  <EmulateController>true</EmulateController>
</Settings>
`

**Comment by abstractoakbow** [comment]
I tried @defjamk12  way to solve issues with laptop interceptor and can also confirm this is a working solution. I was skeptical, assuming you had some USB bluetooth mouse attached or something, but seeing as how your solution worked even for me, that's not the case.
Also to note, my solution of plugging in a mouse failed today while concurrently attaching an external usb hard drive.
Edit:
After 1 hour of play, started to experience bugs. Tapping "W" to take a step forward either resulted in continued running forward for several seconds, or resulted in dropping the input entirely. The same problem was experienced for all inputs.
Closing ps4macro and ps4remote play, and reopening them, and checking the status resulted in this problem
![image](https://user-images.githubusercontent.com/46252000/50617826-b5e38600-0eb4-11e9-8bb8-45a54cfd06e8.png)
with infinite dialogue boxes opening, and the same issue reported in the command console.
Restarting again solved this issued. Repeated implementation of turning on and off bluetooth failed to activate interceptor. However, plugging my mouse back in worked. (also note that removing mouse after remapper is opened does not interfere with play)
